modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java

```diff
@@ -626,6 +631,36 @@ public Boolean isShowSearch() {
 		return false;
 	}
 
+	private void _addAssetTagsFilterLabelItems(
+		LabelItemListBuilder.LabelItemListWrapper labelItemListWrapper) {
+
+		Set<String> assetTagIds = _getSelectedAssetTagIds(_httpServletRequest);
+
+		for (String assetTagId : assetTagIds) {
+			labelItemListWrapper.add(
+				labelItem -> {
+					labelItem.putData(
+						"removeLabelURL",
+						_getRemoveLabelURL(
+							"assetTagId",
+							() -> TransformUtil.transformToArray(
+								assetTagIds,
+								curAssetTagId -> {
+									if (Objects.equals(
+											assetTagId, curAssetTagId)) {
+
+										return null;
+									}
+
+									return curAssetTagId;
+								},
+								String.class)));
```

Comment:
The current `_getRemoveLabelURL` method doesn't seem correct to me:
* What it currently does is just returning an URL with a parameter, not very related to "RemoveLabel"
* If you want to still use it I would hide the logic of `TransformUtil.transformToArray` inside this method
* Or I think is better to transform the set to an array and work like in `_addExtensionFilterLabelItems`, which uses method `ArrayUtil.remove()`

Reply:
> The name of the method is what returns (removeLabelURL) , so the name on this case is correct 🤷🏻‍♀️

You can call the method `_getUrl` and would be still correct. That's my point.

> On _addExtensionFilterLabelItems you are removing an element of the array you are in (on the for) I prefer not to do it.
> Thats why I prefer the use of the TransformUtil.transformToArray( and my idea is to use it on the categories ticket.

`ArrayUtil.remove()` doesn't modify the input array.
Anyway, if you prefer to use `TransformUtil.transformToArray` I would do this work inside the method

Reply:
If I reuse the method as it like with extension:

```
		_getRemoveLabelURL(
				"extension",
			() -> ArrayUtil.remove(extensions, extension)));
```
							
You are ok with it?